caffe2/python/pybind_state.cc

```diff
@@ -1502,6 +1503,27 @@ void addGlobalMethods(py::module& m) {
     new_proto.SerializeToString(&out);
     return py::bytes(out);
   });
+  m.def(
+      "transform_onnxifi",
+      [](const py::bytes& pred_net_str,
+         const std::unordered_map<std::string, std::vector<int>>& shapes,
+         bool debug_builder) -> py::bytes {
+        caffe2::NetDef pred_net;
+        if (!ParseProtoFromLargeString(
+                pred_net_str.cast<std::string>(), &pred_net)) {
```

Comment:
Does it create a copy?

Reply:
Maybe yes. I didn't go into that function. But for here, predict net is usually not big. I think performance wise this should be ok.